### PR TITLE
4 - Make more use of `let_chains`

### PR DESCRIPTION
**Pull request by c410-f3r**
Continuation of #94376.

cc #53667

### CODE REVIEW

File: compiler/rustc_parse/src/parser/mod.rs [hunk 1]
```diff
@@ -97,15 +97,15 @@ macro_rules! maybe_whole {
 #[macro_export]
 macro_rules! maybe_recover_from_interpolated_ty_qpath {
     ($self: expr, $allow_qpath_recovery: expr) => {
-        if $allow_qpath_recovery && $self.look_ahead(1, |t| t == &token::ModSep) {
-            if let token::Interpolated(nt) = &$self.token.kind {
-                if let token::NtTy(ty) = &**nt {
+        if $allow_qpath_recovery
+                    && $self.look_ahead(1, |t| t == &token::ModSep)
+                    && let token::Interpolated(nt) = &$self.token.kind
+                    && let token::NtTy(ty) = &**nt
```
Comment (cjgillot) [inline compiler/rustc_parse/src/parser/mod.rs:103]: The formatting is a bit odd here.

**Review by cjgillot — APPROVED** [review]
Thanks. Just one nit. r=me once resolved.

File: compiler/rustc_parse/src/parser/mod.rs (hunk 1 above)
Comment (rqytda) [reply compiler/rustc_parse/src/parser/mod.rs:103]: Rustfmt stuff and for some reason, #[rustfmt::skip] breaks the build

I can remove this snippet if you want it

Comment (c410-f3r) [reply compiler/rustc_parse/src/parser/mod.rs:103]: Rustfmt stuff and for some reason, #[rustfmt::skip] breaks the build

I can remove this snippet if you want it

Comment (cjgillot) [reply compiler/rustc_parse/src/parser/mod.rs:103]: If rustfmt says so...

Comment (c410-f3r) [reply compiler/rustc_parse/src/parser/mod.rs:103]: Thank you

**Pull request merged**